lms/djangoapps/course_api/views.py

```diff
@@ -2,52 +2,122 @@
-from xmodule.modulestore.django import modulestore
+
+from .api import course_detail, list_courses
 
 
 @view_auth_classes()
-class CourseView(APIView):
+class CourseDetailView(APIView):
     """
-    Course API view
+    View class to describe a course
     """
 
     def get(self, request, course_key_string):
         """
         Request information on a course specified by `course_key_string`.
-            Body consists of a `blocks_url` that can be used to fetch the
-            blocks for the requested course.
+
+        Body consists of the following fields:
+            * `blocks_url`: can be used to fetch the blocks
+            * `course_image`:
+            * `description`:
+            * `end`: Date the course ends
+            * `enrollment_end`: Date enrollment ends
+            * `enrollment_start`: Date enrollment begins
+            * `id`: Course key
+            * `name`: Name of the course
+            * `number`: Catalog number of the course
+            * `org`: Name of the organization that owns the course
+            * `start`: Date the course begins
+            * `start_display`: Readably formatted start of the course
+            * `start_type`: Hint describing how `start_display` is set. One of:
+                  - `"string"`: manually set
+                  - `"timestamp"`: generated form `start` timestamp
+                  - `"empty"`: the start date should not be shown
 
         Arguments:
-            request (HttpRequest)
-            course_key_string
+            request: (HttpRequest)
+            course_key_string: (string) Key for the desired course
+
+        Parameters:
+            username (optional):
+                The username of the specified user whose visible courses we
+                want to see.  Defaults to the current user.
 
         Returns:
             HttpResponse: 200 on success
 
-
         Example Usage:
-
             GET /api/courses/v1/[course_key_string]
             200 OK
 
         Example response:
+            {
+                "blocks_url": "/api/courses/v1/blocks/?course_id=edX%2Ftoy%2F2012_Fall",
+                "course_image": "/c4x/edX/toy/asset/just_a_test.jpg",
+                "description": "A course about toys.",
+                "end": "2015-09-19T18:00:00Z",
+                "enrollment_end": "2015-07-15T00:00:00Z",
+                "enrollment_start": "2015-06-15T00:00:00Z",
+                "id": "edX/toy/2012_Fall",
+                "name": "Toy Course",
+                "number": "toy",
+                "org": "edX",
+                "start": "2015-07-17T12:00:00Z",
+                "start_display": "July 17, 2015",
+                "start_type": "timestamp"
+            }
+        """
+        username = request.query_params.get('username', request.user.username)
+        try:
+            course_key = CourseKey.from_string(course_key_string)
+        except InvalidKeyError:
+            raise NotFound()
```

Comment:
It looks like DRF doesn't like returning NotFounds.  When I changed this to return, I got:

    AssertionError: Expected a `Response`, `HttpResponse` or `HttpStreamingResponse` to be 
    returned from the view, but received a `<class 'rest_framework.exceptions.NotFound'>`